Cleanup

imlucas opened the pull request:
- normalize `.eslintrc`
- `mongodb-js-fmt` is deprecated
- Update `eslint-config-mongodb-js` and `mongodb-js-precommit` for latest rules
- Update `.npmignore` to not publish sourcemaps

```
-rw-r--r--  1 lucas  staff   1.8M Oct 26  1985 node_modules/@mongodb-js/compass-indexes/lib/index.js.map
```

Related: mongodb-js/compass-collection-stats#8

imlucas committed:
Cleanup. See mongodb-js/compass-collection-stats#8

aherlihy commented:
The reason I changed the eslint was because my editor was complaining - there is some error that I searched and searched and found that the solution was to make sure that 

```
 "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "modules": true
    }
}
```

as described here: https://github.com/eslint/eslint/issues/4344#issuecomment-197344872.

The error I was getting before, and am getting again now as a result of these changes:

<img width="961" alt="Screen Shot 2019-05-03 at 9 40 09 AM" src="https://user-images.githubusercontent.com/1045019/57124882-aa7f6900-6d87-11e9-82f7-734c07898c53.png">

Do you know how to fix that using the changes you made? If not can we revert it?

imlucas commented:
@aherlihy ah thanks! I just added back on master. I also made mongodb-js/eslint-config#46 as these type of changes can be done in higher level shared configs and tested there so we don't have so much copy and paste + even more importantly, don't unexpectedly break like this.

I added the `parserOptions` fix to that issue already but if there are other pains happening, plz add to that issue and we can fix them all in one go later in the week.
